src/libs/ajax/User.ts

```diff
@@ -0,0 +1,398 @@
+const getFirstTimeStamp = Utils.memoizeAsync(
+  async (token): Promise<RexFirstTimestampResponse> => {
+    const res = await fetchRex('firstTimestamps/record', _.mergeAll([authOpts(token), { method: 'POST' }]));
+    return res.json();
+  },
+  { keyFn: (...args) => JSON.stringify(args) }
+) as (...args: any[]) => RexFirstTimestampResponse;
+
+export const User = (signal?: AbortSignal) => {
+  return {
+    getStatus: async (): Promise<SamUserRegistrationStatusResponse> => {
+      const res = await fetchSam('register/user/v2/self/info', _.mergeAll([authOpts(), { signal }]));
+      return res.json();
+    },
+
+    profile: {
+      get: async (): Promise<OrchestrationUserProfileResponse> => {
+        const res = await fetchOrchestration('register/profile', _.merge(authOpts(), { signal }));
+        return res.json();
+      },
+
+      // We are not calling Thurloe directly because free credits logic was in orchestration
+      set: async (keysAndValues: {}): Promise<void> => {
+        const blankProfile = {
+          firstName: 'N/A',
+          lastName: 'N/A',
+          title: 'N/A',
+          institute: 'N/A',
+          department: 'N/A',
+          institutionalProgram: 'N/A',
+          programLocationCity: 'N/A',
+          programLocationState: 'N/A',
+          programLocationCountry: 'N/A',
+          pi: 'N/A',
+          nonProfitStatus: 'N/A',
+        };
+        return fetchOrchestration(
+          'register/profile',
+          _.mergeAll([authOpts(), jsonBody(_.merge(blankProfile, keysAndValues)), { signal, method: 'POST' }])
+        );
+      },
+
+      setPreferences: async (body: {}): Promise<void> => {
+        return fetchOrchestration(
+          'api/profile/preferences',
+          _.mergeAll([authOpts(), jsonBody(body), { signal, method: 'POST' }])
+        );
+      },
+
+      preferLegacyFirecloud: async (): Promise<OrchestrationUserPreferLegacyFireCloudResponse> => {
+        return fetchOrchestration('api/profile/terra', _.mergeAll([authOpts(), { signal, method: 'DELETE' }]));
+      },
+    },
+
+    // Returns the proxy group email of the user with the given email
+    getProxyGroup: async (email: string): Promise<string> => {
+      const res = await fetchOrchestration(
+        `api/proxyGroup/${encodeURIComponent(email)}`,
+        _.merge(authOpts(), { signal })
+      );
+      return res.json();
+    },
+
+    getTos: async (): Promise<string> => {
+      const response = await fetchSam('tos/text', _.merge(authOpts(), { signal }));
+      return response.text();
+    },
+
+    acceptTos: async (): Promise<SamUserTosStatusResponse | undefined> => {
+      try {
+        const response = await fetchSam(
+          'register/user/v1/termsofservice',
+          _.mergeAll([authOpts(), { signal, method: 'POST' }, jsonBody('app.terra.bio/#terms-of-service')])
+        );
+        return response.json();
+      } catch (error: any) {
+        if (error.status !== 404) {
```

Comment:
Suggest avoiding `any` wherever possible

```suggestion
      } catch (error: unknown) {
        if (!(error instanceof Response && response.status === 404)) {
```

Same comment for other `catch` blocks.